health-services/project-factory/src/server/api/genericApis.ts

```
@@ -701,28 +666,29 @@
+  const maxLength = 59; // Max length allowed for unique values
 
-  // Iterate over each [key, value] pair in elementCodesMap using forEach
   elementCodesMap.forEach((value: any, key: any) => {
-    let modifiedValue = value.replace(specialCharsRegex, '_').trim(); // Replace special characters and spaces with underscore
-    let modifiedTempValue = modifiedValue; // Store the initial modified value
+    let modifiedValue = value.replace(specialCharsRegex, '_').trim(); // Replace special characters and trim
+    if (modifiedValue.length > maxLength) {
+      logger.info(`Length of ${modifiedValue} is greater than max length ${maxLength}, so it will be truncated to ${maxLength} characters.`);
+    }
+    modifiedValue = modifiedValue.substring(0, maxLength); // Ensure length is at most 59
+    let modifiedTempValue = modifiedValue; // Store the base modified value
```

Biome (1.9.4)

[error] This let declares a variable that is only assigned once.

'modifiedTempValue' is never reassigned.

Safe fix: Use const instead.


(lint/style/useConst)